### PR TITLE
Use opam2

### PR DESCRIPTION
**Pull request by samoht**

### CODE REVIEW

**Comment by samoht** [comment]
I'm testing that PR with https://github.com/mirage/ocaml-conduit/pull/267

File: .travis-ocaml.sh [hunk 1]
```diff
@@ -43,42 +43,12 @@ UBUNTU_TRUSTY=${UBUNTU_TRUSTY:-"0"}
...
-    4.07,1.2.2)
-        OCAML_VERSION=4.02; OCAML_FULL_VERSION=4.07.0
-        ppa=avsm/ocaml42+opam12 ;;
-    *) echo "Unknown OCAML_VERSION=$OCAML_VERSION OPAM_VERSION=$OPAM_VERSION"
-       echo "(An unset OCAML_VERSION used to default to \"latest\", but you must now specify it."
-       echo "Try something like \"OCAML_VERSION=3.12\", \"OCAML_VERSION=4.07\", or see README-travis.md at https://github.com/ocaml/ocaml-ci-scripts )"
-       exit 1 ;;
-  esac
+install_opam2 () {
+  sudo wget https://github.com/ocaml/opam/releases/download/2.0.0-rc3/opam-2.0.0-rc3-x86_64-linux -O /usr/local/bin/opam ;;
```
Comment (kit-ty-kate) [inline .travis-ocaml.sh]: mmh maybe a `sudo chmod +x /usr/local/bin/opam` is going to be missing at some point I guess.

File: .travis-ocaml.sh [hunk 2]
```diff
@@ -43,42 +43,16 @@ UBUNTU_TRUSTY=${UBUNTU_TRUSTY:-"0"}
...
-    4.07,1.2.2)
-        OCAML_VERSION=4.02; OCAML_FULL_VERSION=4.07.0
-        ppa=avsm/ocaml42+opam12 ;;
-    *) echo "Unknown OCAML_VERSION=$OCAML_VERSION OPAM_VERSION=$OPAM_VERSION"
-       echo "(An unset OCAML_VERSION used to default to \"latest\", but you must now specify it."
-       echo "Try something like \"OCAML_VERSION=3.12\", \"OCAML_VERSION=4.07\", or see README-travis.md at https://github.com/ocaml/ocaml-ci-scripts )"
-       exit 1 ;;
-  esac
+install_opam2 () {
+    sudo add-apt-repository ppa:ansible/bubblewrap
```
Comment (rjbou) [inline .travis-ocaml.sh]: Should'nt sandboxing disabled for CI? Besides, for ubuntu, `bwrap` is not yet available on [all versions](https://github.com/ocaml/opam/issues/3424#issuecomment-403782929).

**Comment by samoht** [comment]
It's all green but I am not sure it's actually testing anything.

See https://github.com/ocaml/opam/issues/3463

Also, the PR uncovered a bug with [opam remove -a](https://github.com/ocaml/opam/issues/3461) with opam2, the workaround (removing every package manually) is not pretty but it seems to work.

**Comment by samoht** [comment]
So it seems to compile the packages fine, even when printing the `ccache` error. Not sure what it is about ...

File: .travis-ocaml.sh (hunk 2 above)
Comment (samoht) [reply .travis-ocaml.sh]: It's available on the ubuntu image used by Travis CI. Also recursive sandboxing is available for these images as they are running them in privilege mode.

`.travis-docker.sh` is using the ability to run unprivileged docker images on the Travis CI infrastructure, so has different constraints and different of setting everything up.

**Comment by hannesm** [comment]
I looked through the diff, this looks pretty good.  What is missing is updates of the documentation in (at least) README-travis.md.  I'm a bit worries why travis on this PR fails (during `opam lint`) - is this supposed to happen (i.e. will all users of ocaml-ci-script fail unless their opam files are updated to 2.0 in the respective repositories)?  I'm not sure when the best time is to introduce a breaking change (change of the default version of OPAM).

**Comment by samoht** [comment]
@hannesm I've removed the commit which were using my fork/branch to compile yorick. With that commit, everything was working fine (see the status of f4f2a70). That PR should not bring any visible difference to users (unless it's a new error specific to the use of opam2).

I'll add a bit of documentation.

**Pull request merged**